src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1041,16 +1050,29 @@ impl<'a, 'tcx> DocVisitor for LinkCollector<'a, 'tcx> {
         // Rather than merging all documentation into one, resolve it one attribute at a time
         // so we know which module it came from.
         for (parent_module, doc) in item.attrs.collapsed_doc_value_by_module_level() {
+            if !may_have_doc_links(&doc) {
+                continue;
+            }
             debug!("combined_docs={}", doc);
             // NOTE: if there are links that start in one crate and end in another, this will not resolve them.
             // This is a degenerate case and it's not supported by rustdoc.
             let parent_node = parent_module.or(parent_node);
-            for md_link in markdown_links(&doc) {
+            let mut tmp_links = self
+                .cx
+                .resolver_caches
+                .markdown_links
```

Comment:
Why are these cached? I wouldn't expect markdown parsing to be expensive compared to all the other things rustdoc does. If you look at https://perf.rust-lang.org/compare.html?start=1ec2c136b35bd4660cba778346f8df7daf2ab173&end=cd5dce912ebe17db000b68c530333a49ad85d462&stat=max-rss this also comes with a max-rss hit.

Reply:
See https://github.com/rust-lang/rust/pull/96135#issuecomment-1101475868.

Many things in this area use `String`s while they could benefit from using `Symbol`s instead, maybe that can help with max-rss.